phi/model/openai/chat.py

```diff
@@ -231,75 +306,55 @@ async def ainvoke_stream(self, messages: List[Message]) -> Any:
-        response_content: Optional[str] = response_message.content
-        response_tool_calls: Optional[List[ChatCompletionMessageToolCall]] = response_message.tool_calls
-
-        # -*- Create assistant message
-        assistant_message = Message(
-            role=response_role or "assistant",
-            content=response_content,
-        )
-        if response_tool_calls is not None:
-            assistant_message.tool_calls = [t.model_dump() for t in response_tool_calls]
-
-        # -*- Update usage metrics
-        # Add response time to metrics
-        assistant_message.metrics["time"] = response_timer.elapsed
-        if "response_times" not in self.metrics:
-            self.metrics["response_times"] = []
-        self.metrics["response_times"].append(response_timer.elapsed)
-
-        # Add token usage to metrics
-        response_usage: Optional[CompletionUsage] = response.usage
+        Args:
+            agent_message (Message): The assistant message.
+            response_usage (Optional[CompletionUsage]): The response usage.
+        """
         if response_usage:
             prompt_tokens = response_usage.prompt_tokens
             completion_tokens = response_usage.completion_tokens
             total_tokens = response_usage.total_tokens
 
             if prompt_tokens is not None:
-                assistant_message.metrics["prompt_tokens"] = prompt_tokens
+                agent_message.metrics["prompt_tokens"] = prompt_tokens
+                agent_message.metrics["input_tokens"] = prompt_tokens
                 self.metrics["prompt_tokens"] = self.metrics.get("prompt_tokens", 0) + prompt_tokens
-                assistant_message.metrics["input_tokens"] = prompt_tokens
                 self.metrics["input_tokens"] = self.metrics.get("input_tokens", 0) + prompt_tokens
             if completion_tokens is not None:
-                assistant_message.metrics["completion_tokens"] = completion_tokens
+                agent_message.metrics["completion_tokens"] = completion_tokens
+                agent_message.metrics["output_tokens"] = completion_tokens
                 self.metrics["completion_tokens"] = self.metrics.get("completion_tokens", 0) + completion_tokens
-                assistant_message.metrics["output_tokens"] = completion_tokens
                 self.metrics["output_tokens"] = self.metrics.get("output_tokens", 0) + completion_tokens
             if total_tokens is not None:
-                assistant_message.metrics["total_tokens"] = total_tokens
+                agent_message.metrics["total_tokens"] = total_tokens
                 self.metrics["total_tokens"] = self.metrics.get("total_tokens", 0) + total_tokens
 
-        # -*- Add assistant message to messages
-        messages.append(assistant_message)
-        assistant_message.log()
-
-        # -*- Parse and run tool calls
-        if assistant_message.tool_calls is not None and self.run_tools:
+    def _handle_tool_calls(
```

Comment:
i think its okay for these functions to be without a leading `_` now, wdyt? 

`def handle_tool_calls()`

Reply:
Its a python best practice:

- The method or variable is intended for internal use within the class or module.

- It's a "weak" form of name mangling - it doesn't prevent access from outside the class, but it's a strong hint that the method should not be called directly.

Reply:
good one :)